lib/features/coin_flow/screens/search_for_coin_screen.dart

```diff
@@ -1,43 +1,95 @@
 import 'package:givt_app_kids/shared/widgets/floating_action_button.dart';
 import 'package:go_router/go_router.dart';
 
 class SearchForCoinScreen extends StatelessWidget {
-  const SearchForCoinScreen({super.key});
+  const SearchForCoinScreen({super.key, required this.mediumId});
+  final String mediumId;
 
   @override
   Widget build(BuildContext context) {
-    return BlocBuilder<SearchCoinCubit, SearchCoinState>(
-      builder: (BuildContext context, state) {
-        return Scaffold(
-          backgroundColor: const Color(0xFFEEEDE4),
-          body: state is SearchCoinAnimationState
-              ? const SearchingForCoinPage()
-              : const CoinFoundedPage(),
-          floatingActionButtonLocation:
-              FloatingActionButtonLocation.centerFloat,
-          floatingActionButton: state is SearchCoinFoundedState
-              ? FloatingActoinButton(
-                  text: "Assign the coin",
-                  onPressed: () {
-                    AnalyticsHelper.logEvent(
-                        eventName: AmplitudeEvent.buttonPressed,
-                        eventProperties: {
-                          'button_name': 'Assign the coin',
-                          'formatted_date': DateTime.now().toIso8601String(),
-                          'screen_name': Pages.searchForCoin.name,
-                        });
+    final coinCubit = context.read<SearchCoinCubit>();
+    return BlocConsumer<SearchCoinCubit, SearchCoinState>(
+      listener: (context, coinState) {
+        if (coinState.status == CoinAnimationStatus.animating) {
+          context
+              .read<OrganisationDetailsCubit>()
+              .getOrganisationDetails(mediumId);
+        }
+      },
+      builder: (context, coinState) {
+        return BlocConsumer<OrganisationDetailsCubit, OrganisationDetailsState>(
+          listener: (context, orgState) async {
+            if (orgState is OrganisationDetailsSetState) {
+              log("Organisation is set: ${orgState.organisation.name}");
+              AnalyticsHelper.logEvent(
+                  eventName: AmplitudeEvent.nfcScanned,
+                  eventProperties: {
+                    'goal_name': orgState.organisation.name,
+                  });
+              // checking if the request took less than the animation duration
+              if (coinState.stopwatch.elapsedMilliseconds <
+                  SearchCoinCubit.searchDuration.inMilliseconds) {
+                await Future.delayed(
+                  SearchCoinCubit.searchDuration -
+                      Duration(
+                        milliseconds: coinState.stopwatch.elapsedMilliseconds,
+                      ),
+                );
+              }
+              coinCubit.stopAnimation(coinState);
+            }
+            if (orgState is OrganisationDetailsErrorState) {
+              // checking if the request took less than the animation duration
+              ScaffoldMessenger.of(context).showSnackBar(
```

Comment:
we need to think more about error handling for the coin flow, since while debugging I got an error from fetching org details, and I've seen the snackbar error message, but the coin animation was animated forever.

Reply:
Something like a retry button, on it 🏃

Reply:
Added CoinErrorPage

Reply:
the error handling works well now, however shall we design this page properly as well?